profiling/src/php_ffi.h

```diff
@@ -0,0 +1,133 @@
+#include <SAPI.h>
+#include <Zend/zend_extensions.h>
+#include <Zend/zend_modules.h>
+#include <php.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+#include <ext/standard/info.h>
+
+/**
+ * Represents a non-owning slice of chars. The order here matches GCC std::span,
+ * which changed in v11 to be ptr-size instead of size-ptr to match iovec on
+ * most platforms.
+ *
+ * Please do not use a null pointer; instead use an empty string with a `size`
+ * of 0.
+ *
+ * todo: unify with ../components/string_view
+ */
+typedef struct datadog_php_str {
+    const char *ptr;
+    size_t size;
+} datadog_php_str;
+
+/* C11 allows a duplicate typedef provided they are the same, so this should be
+ * fine as long as we compile with C11 or higher.
+ */
+typedef ZEND_RESULT_CODE zend_result;
+
+/**
+ * Returns macro expansion of ZEND_EXTENSION_BUILD_ID, which bindgen cannot
+ * currently handle.
+ */
+const char *datadog_extension_build_id(void);
+
+/**
+ * Returns macro expansion of ZEND_MODULE_BUILD_ID, which bindgen cannot
+ * currently handle.
+ */
+const char *datadog_module_build_id(void);
+
+/**
+ * Represents a log level filter setting. Store these in uintptr_t, as the Rust
+ * type is repr(usize) aka uintptr_t.
+ * todo: unify with ../component/log
+ */
+typedef enum datadog_php_profiling_log_level {
+    DATADOG_PHP_PROFILING_LOG_LEVEL_OFF = 0,
+    DATADOG_PHP_PROFILING_LOG_LEVEL_ERROR,
+    DATADOG_PHP_PROFILING_LOG_LEVEL_WARN,
+    DATADOG_PHP_PROFILING_LOG_LEVEL_INFO,
+    DATADOG_PHP_PROFILING_LOG_LEVEL_DEBUG,
+    DATADOG_PHP_PROFILING_LOG_LEVEL_TRACE,
+} datadog_php_profiling_log_level;
```

Comment:
likewise, I don't see this being used at all; I just see `log::LevelFilter` being transmute()d into a uintptr_t in read_env().

Reply:
Yes, this was also used for the `.ini` handling, but it somehow lost its `repr(C)` along the way.